core/deployment/src/main/java/io/quarkus/deployment/recording/PropertyUtils.java

```diff
@@ -10,17 +9,12 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-import java.util.function.Function;
 
 final class PropertyUtils {
 
-    static final ConcurrentMap<Class<?>, Property[]> CACHE = new ConcurrentHashMap<>();
-
-    private static final Function<Class<?>, Property[]> FUNCTION = new Function<Class<?>, Property[]>() {
+    private static final ClassValue<Property[]> CACHE = new ClassValue<Property[]>() {
```

Comment:
+1

Reply:
N.B. I'm not sure the original code would leak, but this seems much safer to me. I'm referring to classloader leaks in case there's a recorder triggered as this is only capturing properties of recorders so I don't think it would happen.

Reply:
Yup